src/adapter/src/catalog.rs

```diff
@@ -2767,8 +2771,16 @@ impl<S: Append> Catalog<S> {
                 _ => unreachable!("can't migrate id: {id}"),
             };
 
-            // Generate value to update fingerprint and global ID persisted mapping.
-            if let Some(fingerprint) = id_fingerprint_map.get(&id) {
+            let name = self.resolve_full_name(entry.name(), None);
+            info!("migrating {name} from {id} to {new_id}");
+
+            // Generate value to update fingerprint and global ID persisted mapping for system objects.
+            // Not every system object has a fingerprint, like introspection source indexes.
+            if let Some(fingerprint) = id_fingerprint_map.get(id) {
+                assert!(
```

Comment:
No, users can still create indexes on system objects. They are just tracked in a separate collection in the stash, and don't have fingerprints, so they aren't included in this list. Since this list is just used to update the system object collection in the stash.